plugin/src/main/java/com/gtnewhorizons/retrofuturagradle/modutils/ModUtils.java

```diff
@@ -105,6 +107,24 @@ public ModUtils(Project project, MinecraftExtension mcExt, MinecraftTasks minecr
                     "Apply MCP decompiler cleanup to the main source set, doing things like replacing numerical OpenGL constants with their names");
         });
 
+        project.getTasks().register("migrateMappings", MigrateMappingsTask.class, task -> {
+            task.setGroup(TASK_GROUP_USER);
+            task.setDescription("Migrate main source set to a new set of mappings");
+            task.dependsOn("extractMcpData", "extractForgeUserdev", "packagePatchedMc", "injectTags");
+            task.getSourceSrg()
+                    .set(mcpTasks.getTaskGenerateForgeSrgMappings().flatMap(GenSrgMappingsTask::getInputSrg));
+            task.getSourceFieldsCsv()
+                    .set(mcpTasks.getTaskGenerateForgeSrgMappings().flatMap(GenSrgMappingsTask::getFieldsCsv));
+            task.getSourceMethodsCsv()
+                    .set(mcpTasks.getTaskGenerateForgeSrgMappings().flatMap(GenSrgMappingsTask::getMethodsCsv));
+            ConfigurableFileCollection cp = task.getCompileClasspath();
+            cp.from(project.getConfigurations().getByName("compileClasspath"));
+            cp.from(project.files(project.getTasks().named("packagePatchedMc", Jar.class)));
+            cp.from(
+                    project.getExtensions().getByType(JavaPluginExtension.class).getSourceSets()
+                            .getByName("injectedTags").getOutput());
```

Comment:
Not entirely sure if I'm supposed to use `.named` here, when I tried using it it gave me an inscrutable compiler error so I went with getting it directly.